ignite/contrib/handlers/tensorboard_logger.py

```diff
@@ -0,0 +1,305 @@
+try:
+    from tensorboardX import SummaryWriter
+except ImportError:
+    raise RuntimeError("This contrib module requires tensorboardX to be installed.")
+
+from ignite.engine import Engine, Events
+from torch import nn as nn
+import os
+
+
+class TensorboardLogger(object):
+    """
+    TensorBoard handler to log metrics related to training, validation and model.
+
+    Examples:
+
+        Create a TensorBoard summary that visualizes metrics related to training,
+        validation, and model parameters, by simply attaching the handler to your engine.
+
+        ..code-block:: python
+
+            tbLogger = TensorboardLogger()
+
+            tbLogger.attach(engine=trainer,
+                            mode='epoch',
+                            model=model,
+                            dataloader=data,
+                            use_metrics=True,
+                            state_keys=['reward'],
+                            histogram_freq=1,
+                            write_grads=True)
+
+            tbLogger.attach_evaluator(engine=evaluator,
+                                      use_metrics=True,
+                                      state_keys=['reward'])
+
+    Note:
+        When adding tensorboard event handler to an engine, it is recommended that the evaluator
+        is run before the code block above.
+
+        Depending on the engines this callback is attached to, SummaryWriter's are saved in self.writer
+        which is a dictionary of keys 'train' or 'eval. If custom functions are needed, use
+        tbLogger.writer['train'].add_*.
+    """
+
+    def __init__(self, log_dir=None):
+        self.attached = False
+        if not log_dir:
+            import socket
+            from datetime import datetime
+            current_time = datetime.now().strftime('%b%d_%H-%M-%S')
+            log_dir = os.path.join('runs', current_time + '_' + socket.gethostname())
+        self.log_dir = log_dir
+        self.writer = {}
+
+    def _create_writer(self, mode):
+        self.writer[mode] = SummaryWriter(log_dir=os.path.join(self.log_dir, mode))
+
+    def _close(self, engine):
+        """
+        Closes Summary Writer
+        """
+        for key, writer in self.writer.items():
+            writer.close()
+
+    def _write_graph(self, engine, model, dataloader):
+        """
+        Create computational graph of model.
+
+        Args:
+            model (nn.Module): model to be written
+            dataloader (torch.utils.DataLoader): data loader for training data
+        """
+        writer = self.writer['train']
+        x, y = next(iter(dataloader))
```

Comment:
Assuming that seeds are set manually at the beginning of an experiment, this line of code can break reproducibility. Running the same experiment with `write_graph=True` and `write_graph=False` can result in different batch data passed to the engines `process_function` because the dataloader is called once here. I personally do not use the `write_graph` option but if `batch_size` or number of samples passed to the model to write a graph is unimportant, one can use 
```
x, y = dataloader.collate_fn(dataloader.dataset.__getitem__(0))
```
instead?